File: youBookingBack/src/main/java/com/example/app/Entity/Reservation.java
package com.example.app.Entity;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private Long id_reservation;
    private double totalPrice;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate ;
    private Status status;
    @ManyToOne
    @JsonBackReference
    private Room room;
    @ManyToOne(fetch = FetchType.EAGER)
    private appUser client;
}

File: youBookingFront/src/app/model/reservation.ts
import * as moment from "moment";
import {Room} from "./room";
import {User} from "./user";


export class Reservation {

  id_reservation !: number;
  totalPrice !: number;
  startDate !: Date;
  endDate !: Date;
  status !: string;
  room !: Room;
  client !: User;



 /* constructor(id_reservation: number,totalPrice: number, startDate: Date, endDate: Date, status: string, room: Room) {
    this.id_reservation = totalPrice;
    this.totalPrice = totalPrice;
    this.startDate = startDate;
    this.endDate = endDate;
    this.status = status;
    this.check_in = moment(this.startDate).toDate();
    this.check_out = moment(this.endDate).toDate();
    this.room = room;
  }*/

  //public Reservation() {}

  getRoom() {
    return this.room;
  }
}
